### PR TITLE
Document and initiate new logging strategy to return log file content.

### LINKED ISSUE #647: Document and initiate new logging strategy to return log file content.
The current logging design is to return logging URLs that direct the user to the UI of the logging system. This is not what is desired.

The logging URLs should instead be used by the API server to contact the logging system and retrieve the log file content. This issue will update the documentation and design logging, add some resources to be utilized in the implementation, and update the logging integrations to be able to generate URLs to return the log content.

This issue will then spawn other issues to implement changes in both the sink, the API server, and the CLI.

**Acceptance criteria**

- Splunk and Elasticsearch integrations updated to log full log line instead of parsed line.
- Empty `ConfigMap` named "kubearchive-logging"
- Empty `Secret` named "kubearchive-logging"
- Documentation updated with `ConfigMap` examples for Splunk and Elasticsearch.
- Logging design document updated with changes.

### PR DESCRIPTION
**Pull request by ggallen**
## Which Issue(s) this Pull Request Resolves

Resolves #647

## Release Note

```release-note
NONE
```

### CODE REVIEW

File: docs/modules/ROOT/pages/design/logging-integration.adoc [hunk 1]
```diff
@@ -159,8 +168,8 @@ The sink should first delete any existing entries in the `log_url` table for the
 `Pod` being archived. Earlier archival requests may have already created records
 in the `log_url` table, and they should be removed to avoid duplicates.
 
-The sink will also need to mount and use the `ConfigMap` for logging as detailed
-in the <<Configuration>> section. Additionally, the sink will overwrite the ConfigMap
+The sink will also need to mount and use the `kubearchive-logging` `ConfigMap`
```
Comment (skoved) [inline docs/modules/ROOT/pages/design/logging-integration.adoc:171]:
```suggestion
The sink will mount and use the `kubearchive-logging` `ConfigMap`
```

File: docs/modules/ROOT/pages/design/logging-integration.adoc [hunk 2]
```diff
@@ -170,38 +179,25 @@ The CLI will implement a `logs` command similar to `kubectl logs`.
...
 
 The CLI will have to traverse the owner references to gather all of the logs associated
 with the given resource. Note that logs could be queried for any resource. Most
 resource will not have any logs associated with them or their descendents, but queries on
 logs for `TaskRuns` and `Pods` are possible.
 
-=== API
-
-The API needs to providate a mechanism similar to the CLI for retrieving log URLs.
+The CLI will also have to perform post-processing on the output from the response to
```
Comment (skoved) [inline docs/modules/ROOT/pages/design/logging-integration.adoc:196]:
```suggestion
The CLI will perform post-processing on the output from the response to
```

File: docs/modules/ROOT/pages/design/logging-integration.adoc [hunk 3]
```diff
@@ -170,38 +179,25 @@ The CLI will implement a `logs` command similar to `kubectl logs`.
...
-kind: ConfigMap
-metadata:
-  name: kubearchive-config
-  namespace: kubearchive
-data:
-  logging-configmap-name: kubearchive-splunk
-----
-The sink should check this `ConfigMap`, and if it exists and the `logging-configmap-name` key
-is set, it should mount that `ConfigMap` and use it for generating log URLs.
+The API needs to providate a mechanism similar to the CLI for retrieving log file contents.
```
Comment (skoved) [inline docs/modules/ROOT/pages/design/logging-integration.adoc:203]:
```suggestion
The API will provide a mechanism similar to the CLI for retrieving log file contents.
```

**Review by skoved — APPROVED** [review]
Found a few spots where I think the phrasing can be made more clear, but otherwise lgtm :+1:

File: docs/modules/ROOT/pages/design/logging-integration.adoc (hunk 1 above)
Comment (ggallen) [reply docs/modules/ROOT/pages/design/logging-integration.adoc]: Fixed.

File: docs/modules/ROOT/pages/design/logging-integration.adoc (hunk 2 above)
Comment (ggallen) [reply docs/modules/ROOT/pages/design/logging-integration.adoc]: Fixed.

File: docs/modules/ROOT/pages/design/logging-integration.adoc (hunk 3 above)
Comment (ggallen) [reply docs/modules/ROOT/pages/design/logging-integration.adoc]: Fixed.

**Review by rh-hemartin — APPROVED** [review]

**Pull request merged**